Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDrawingArea.mm

```diff
@@ -526,32 +519,36 @@
 
 void RemoteLayerTreeDrawingArea::scheduleRenderingUpdateTimerFired()
 {
-    triggerRenderingUpdate();
-    m_isScheduled = false;
+    startRenderingUpdateTimer();
 }
 
-bool RemoteLayerTreeDrawingArea::scheduleRenderingUpdate()
+void RemoteLayerTreeDrawingArea::scheduleRenderingUpdate(ScheduleRenderingUrgency urgency)
 {
-    if (m_isScheduled)
-        return true;
+    if (m_updateRenderingIsPending)
```

Comment:
Is it OK that an "as soon as possible" request can't trump a "next suitable time" one?

Reply:
That's a really good point, it's almost certain that we're subtly relying on that for performance in certain situations.

Reply:
Ugh, fixing that added yet another subtle race.

Updated further with a state machine, hopefully all the asserts pass now.